script.js

```diff
@@ -0,0 +1,164 @@
+        const phoneNumber=document.getElementById("phonenumber").value;
+        const password=document.getElementById("password").value;
+        let encriptedPassword = crypt.encrypt(password);
+        if (!usernameValid(username)){
+            document.getElementById("username").style.borderColor="red";
+            alert("Name should contain atleast 3 letters and numbers are not allowed");
+            document.getElementById("username").style.borderColor="white";
+            return false;
+        }
+        if (!emailValid(email) || !emailNotExist(email)){
+            document.getElementById("email").style.borderColor="red";
+            alert("invalid email");
+            document.getElementById("email").style.borderColor="white";
+            return false;
+        }
+        if (!passwordValid(password)){
+            document.getElementById("password").style.borderColor="red";
+            alert("password should contain atlest 8characters including one uppercase,number and special character");
+            document.getElementById("password").style.borderColor="white";
+            return false;
+        }
+        if(!phoneNumberValid(phoneNumber) ||!phonenumNotExist(phoneNumber) ){
+            document.getElementById("phonenumber").style.borderColor="red";
+            alert("phone number should contain 10 numbers");
+            document.getElementById("phonenumber").style.borderColor="white";
+            return false;
+        }
+        addDeatails(username,email,phoneNumber,encriptedPassword);
+        document.getElementById("signup").reset();
+        return true;
+    }
+
+    function validateDetails(event){
+        event.preventDefault();
+        const existingusername=document.getElementById("existingusername").value;
+        const existingPassword=document.getElementById("existingPassword").value;
+        const userExist=userDetails.filter(obj=>obj.userEmail===existingusername);
+        const decriptPassword=crypt.decrypt(userExist[0].userPassword);
+        if (decriptPassword === existingPassword){
+            const user=userExist[0].userName;
+            const idString=userExist[0].id;
+            const userNameid="userName"+idString;
+            setCookie(userNameid,user,1);
+            document.getElementById("login").reset();
+            window.location.href = "https://aiswarya-a-m.github.io/login/dashboard.html";   
+        }
+        else{
+            alert("password is incorrect");
+        }
+    }
+    function usernameValid(username){
+        const usernamePattern=/^[a-zA-Z ]{3,30}$/;
+        if(usernamePattern.test(username)){
+            return true;
+        }else{
+            return false;
+        }
+    }
+    function emailValid(email){
+        const emailPattern=/^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+        if(emailPattern.test(email)){
+            return true;
+        }else{
+            return false;
+        }
```

Comment:
`test()` method on a regex pattern will return either `true` or `false`. so return that instead. of this explicit checking.

```suggestion
        return emailPattern.test(email)
```